modules/effects/src/map_to_action.ts

```diff
@@ -0,0 +1,168 @@
+  // action is the action that was completed
+  complete?: (count: number, action: Input) => CompleteAction;
+  // Optional flattening operator
+  operator?: <InputAction, OutputAction>(
+    project: (input: InputAction, index: number) => Observable<OutputAction>
+  ) => OperatorFunction<InputAction, OutputAction>;
+  // Optional unsubscribe action provider
+  // count is the number of actions project emitted before unsubscribing
+  // action is the action that was unsubscribed from
+  unsubscribe?: (count: number, action: Input) => UnsubscribeAction;
+}
+
+/**
+ * Wraps project fn with error handling making it safe to use in Effects.
+ * Takes either config with named properties that represent different possible
+ * callbacks or project/error callbacks that are required.
+ */
+export function mapToAction<
+  Input,
+  OutputAction extends Action,
+  ErrorAction extends Action
+>(
+  project: (action: Input, index: number) => Observable<OutputAction>,
+  error: (error: any, action: Input) => ErrorAction
+): (source: Observable<Input>) => Observable<OutputAction | ErrorAction>;
+export function mapToAction<
+  InputAction extends Action,
+  OutputAction extends Action,
+  ErrorAction extends Action,
+  CompleteAction extends Action = never,
+  UnsubscribeAction extends Action = never
+>(
+  config: MapToActionConfig<
+    InputAction,
+    OutputAction,
+    ErrorAction,
+    CompleteAction,
+    UnsubscribeAction
+  >
+): (
+  source: Observable<InputAction>
+) => Observable<
+  OutputAction | ErrorAction | CompleteAction | UnsubscribeAction
+>;
+export function mapToAction<
+  Input,
+  OutputAction extends Action,
+  ErrorAction extends Action,
+  CompleteAction extends Action = never,
+  UnsubscribeAction extends Action = never
+>(
+  /** Allow to take either config object or project/error functions */
+  configOrProject:
+    | MapToActionConfig<
+        Input,
+        OutputAction,
+        ErrorAction,
+        CompleteAction,
+        UnsubscribeAction
+      >
+    | ((action: Input, index: number) => Observable<OutputAction>),
+  errorFn?: (error: any, action: Input) => ErrorAction
+): (
+  source: Observable<Input>
+) => Observable<
+  OutputAction | ErrorAction | CompleteAction | UnsubscribeAction
+> {
+  const { project, error, complete, operator, unsubscribe } =
+    typeof configOrProject === 'function'
+      ? {
+          project: configOrProject,
+          error: errorFn!,
+          operator: concatMap,
+          complete: undefined,
+          unsubscribe: undefined,
+        }
+      : { ...configOrProject, operator: configOrProject.operator || concatMap };
+
+  type ResultAction =
+    | OutputAction
+    | ErrorAction
+    | CompleteAction
+    | UnsubscribeAction;
+  return source =>
+    defer(
+      (): Observable<ResultAction> => {
+        const subject = new Subject<UnsubscribeAction>();
+        return merge(
+          source.pipe(
+            operator((action, index) =>
```

Comment:
`input` would be a better descriptor than `action` because it could be something that isn't just an action coming in before you get to this operator.